spark_pipeline_framework/transformers/fhir_sender/v1/fhir_sender.py

```diff
@@ -299,6 +337,9 @@ def _transform(self, df: DataFrame) -> DataFrame:
         )
 
         num_partitions: Optional[int] = self.getOrDefault(self.num_partitions)
+        enable_repartitioning: bool = (
+            True if num_partitions else self.getOrDefault(self.enable_repartitioning)
+        )
```

Comment:
sounds unintuitive when enable_repartitions would be set to False but num_partitions has some value. Check what cases makes sense with enable_repartitioning flag. For eg, if this flag is False and partition_by_column_name has a value, what should be done?
Add asserts if two parameters together doesn't make sense and so on.

Reply:
we can use (partition_by_column_name or num_partitions) as an assert statement in case we're getting `enable_repartitioning=False`. 
We do not need `batch_size` for above scenario as it has different role in `run_synchronously` and it has nothing to do with repartitioning. For a case where desired_partitions are set to `math.ceil(row_count / batch_size)`, it will only work when we're getting  `enable_repartitioning=True`